xcdat/regridder/regrid2.py

```diff
@@ -229,48 +234,87 @@ def _build_dataset(
     input_grid: xr.Dataset,
     output_grid: xr.Dataset,
 ) -> xr.Dataset:
-    input_data_var = ds[data_var]
+    """Build a new xarray Dataset with the given output data and coordinates.
 
-    output_coords: dict[str, xr.DataArray] = {}
-    output_data_vars: dict[str, xr.DataArray] = {}
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The input dataset containing the data variable to be regridded.
+    data_var : str
+        The name of the data variable in the input dataset to be regridded.
+    output_data : np.ndarray
+        The regridded data to be included in the output dataset.
+    input_grid : xr.Dataset
+        The input grid dataset containing the original grid information.
+    output_grid : xr.Dataset
+        The output grid dataset containing the new grid information.
 
-    for dim in input_data_var.dims:
-        dim = str(dim)
+    Returns
+    -------
+    xr.Dataset
+        A new dataset containing the regridded data variable with updated
+        coordinates and attributes.
+    """
+    dv_input = ds[data_var]
 
-        try:
-            axis_name = [
-                cf_axis for cf_axis, dims in ds.cf.axes.items() if dim in dims
-            ][0]
-        except IndexError as e:
-            raise ValueError(
-                f"Could not determine axis name for dimension {dim}"
-            ) from e
-
-        if axis_name in ["X", "Y"]:
-            output_coords[dim] = output_grid.cf[axis_name]
-        else:
-            output_coords[dim] = input_data_var.cf[axis_name]
```

Comment:
Your logic looped over input data variable dims and checks if the dim name is in `ds.cf.axes` (must have `"axis"` attr set).